File: javascript_example/run_api_call.js
const fs = require('fs');
const path = require("path")
const csv_converter = require('json-2-csv')
const fetch = require('node-fetch');

global.fetch = fetch
global.Headers = fetch.Headers;

const SERVICE_URI = process.env.SERVICE_URI;// <----- UPDATE THIS VALUE IN YOUR ENV VARIABLES
const API_KEY = process.env.STATE_API_KEY;// <----- UPDATE THIS VALUE IN YOUR ENV VARIABLES

const example_jsons = path.resolve(__dirname, "../examples/input_samples/json_format/example1.json"); // <----- UPDATE THIS VALUE for various example jsons

if (!SERVICE_URI || SERVICE_URI === "" || !API_KEY || API_KEY === "") {
  console.log("Environment variables SERVICE_URI and/or API_KEY are not set. Please set the local environment variables to proceed.");
} else {
  fs.readFile(example_jsons, (err, data) => { 
    if (err) throw err; 
  
    var myHeaders = new Headers();
      myHeaders.append("Content-Type", "application/json");
      myHeaders.append("apikey", API_KEY); 
      myHeaders.append("Accept", "application/json");  
   
    var raw = JSON.stringify(JSON.parse(data));
    var requestOptions = {
      method: 'POST',
      headers: myHeaders,
      body: raw
    };
    fetch(SERVICE_URI, requestOptions)
      .then(response => response.text())
      .then(result => {
        // raw console output
        if(JSON.parse(result).result) {
          var output_response = JSON.parse(result).result;
          console.log(output_response);
  
          var jsonpath = './output.json';
          var csvpath ='./output.csv';
          // write to json file
          fs.writeFile(jsonpath, JSON.stringify(output_response,null,2), (err) => {
            if (err) throw err;
            console.log("JSON File has been created");
          });          
          // write to csv file
          csv_converter.json2csv(output_response, (err, csv) => {
            if (err) throw err;
            fs.writeFile(csvpath, csv, (err) => {
              if (err) {
                console.error(err);
                return;
              }
              console.log("CSV File has been created");
            });
          });
        } else {
          console.log(result);
        }
      }).catch(error => console.log('error', error));
  });
}

File: python_example/run_api_call.py
import requests
import json
import csv 
import os
from pathlib import Path

# Load our values in from the env variables - update these as needed.
SERVICE_URI = os.environ.get('SERVICE_URI')
API_KEY = os.environ.get('STATE_API_KEY')

if(SERVICE_URI is None or len(SERVICE_URI) == 0 or API_KEY is None or len(API_KEY) == 0):
  print('Environment variables SERVICE_URI and/or API_KEY are not set. Please set the local environment variables to proceed.')
else:
  p = Path()
  example_jsons = p.resolve().parent / 'examples/input_samples/json_format/example1.json' # <--- changes this to load different query examples

  # read your JSON-wrapped SQL query payload from a file
  with open(example_jsons, 'r') as iqies_query:
    payload = str(json.loads(iqies_query.read())).replace("'", '\"')

  headers = {
    'apikey': API_KEY, 
    'Content-Type': 'application/json'
  }

  response = requests.request("POST", SERVICE_URI, headers=headers, data = payload)

  query_output = response.json().get('result', {})  

  if(query_output):
    ### WRITE TO CONSOLE ###
    print(json.dumps(query_output,indent=4))

    ### WRITE TO JSON FILE ###
    with open('output.json', 'w') as iqies_query_results:
        json.dump(query_output, iqies_query_results, sort_keys=True, indent=4)
        print('JSON File has been created')

    ### WRITE TO CSV FILE ###

    # open a file for writing 
    with open('output.csv', 'w') as data_file:
        # create the csv writer object 
        csv_writer = csv.writer(data_file)
        csv_writer.writerow(query_output[0].keys()) # Write the keys to the headers row
        for a in query_output: 
            csv_writer.writerow(a.values()) # Writing data of CSV file 
        print('CSV File has been created')